test/erc20Gateway.ts

```diff
@@ -418,6 +459,89 @@ describe('ERC20 Gateway', async () => {
             expect(messageSentEvent.args.data).to.equal(messageData);
             expect(messageSentEvent.args.amount).to.equal(0);
         });
+
+        it('Should be able to deposit tokens with data', async () => {
+            const gatewayBalance = await env.token.balanceOf(env.fuelERC20Gateway.address);
+            expect(
+                (await env.fuelERC20Gateway.tokensDeposited(tokenAddress, fuelTokenTarget1)).add(
+                    await env.fuelERC20Gateway.tokensDeposited(tokenAddress, fuelTokenTarget2)
+                )
+            ).to.be.equal(gatewayBalance);
+
+            // Deposit to fuelTokenTarget1
+            const toAddress = randomBytes32();
+            const depositData = [3, 2, 6, 9, 2, 5];
+            const depositAmount = 85;
+            await expect(
+                env.fuelERC20Gateway.depositWithData(
+                    toAddress,
+                    tokenAddress,
+                    fuelTokenTarget1,
+                    depositAmount,
+                    depositData
+                )
+            ).to.not.be.reverted;
+            expect(await env.token.balanceOf(env.fuelERC20Gateway.address)).to.be.equal(
+                gatewayBalance.add(depositAmount)
+            );
+
+            // Verify MessageSent event to l2contract
+            const messageData = computeMessageData(
+                fuelTokenTarget1,
+                tokenAddress.split('0x').join('0x000000000000000000000000'),
```

Comment:
Why split at `0x` and then join it again?

Reply:
This left-pads an eth address to 32 bytes.